### PR TITLE
Abstract creditor details component

### PR DESCRIPTION
**Pull request by Arnabsubedi233**
### Jira link

See [https://tools.hmcts.net/jira/browse/PO-2128](https://tools.hmcts.net/jira/browse/PO-2128)

### Change description

- Adding the abstract creditor base component
### Testing done

- unit tested
- locally tested
### Security Vulnerability Assessment ###

**CVE Suppression:** Are there any CVEs present in the codebase (either newly introduced or pre-existing) that are being intentionally suppressed or ignored by this commit?
  * [ ] Yes
  * [x] No

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: projects/opal-frontend-common/components/abstract/abstract-creditor-details-base/abstract-creditor-details-base.component.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+})
+export abstract class AbstractCreditorDetailsBaseComponent<THeader, TTab extends { version: string | null }>
+  extends AbstractAccountSummaryBaseComponent<THeader, TTab>
+  implements OnInit, OnDestroy
+{
+  protected abstract readonly headerDataRouteKey: string;
+  protected abstract readonly defaultActiveTab: string;
+  protected abstract readonly latestBannerMessage: string;
+
+  public accountId: number = Number(this.activatedRoute.snapshot.paramMap.get('accountId'));
```
Comment (iamfrankiemoran) [inline projects/opal-frontend-common/components/abstract/abstract-creditor-details-base/abstract-creditor-details-base.component.ts]: [P1] This hard-codes the shared base to an `accountId` route param. That matches the current `opal-frontend` creditor details pages, but it makes the common-lib abstraction less reusable and can silently produce `NaN` for any consumer with a different route shape. Can we make account id resolution an abstract method, for example `protected abstract getAccountId(): number`, or at least make the route param key configurable?

File: projects/opal-frontend-common/components/abstract/abstract-creditor-details-base/abstract-creditor-details-base.component.ts [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+   * @returns The transformed tab data with the original tab type preserved.
+   */
+  protected override transformTabData<T extends TTab>(data: T): T {
+    return this.transformPayload(data);
+  }
+
+  /**
+   * Refreshes the current creditor summary page and updates store state after the latest header is loaded.
+   */
+  public override refreshPage(): void {
```
Comment (iamfrankiemoran) [inline projects/opal-frontend-common/components/abstract/abstract-creditor-details-base/abstract-creditor-details-base.component.ts:96]: [P1] This moves quite a lot of fines account page workflow into the common base: clearing version mismatch, refreshing by account id, setting a success banner, updating `accountData`, and then relying on the inherited refresh fragment flow. The abstraction works for the current creditor pages, but it feels tightly coupled to the `opal-frontend` store/banner lifecycle. Can we keep this orchestration in the app-specific adapter, or pass it in as a smaller strategy/config so the common class stays focused on generic account-summary behaviour?

File: projects/opal-frontend-common/components/abstract/abstract-creditor-details-base/abstract-creditor-details-base.component.ts [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+   */
+  protected abstract clearSuccessMessage(): void;
+
+  /**
+   * Transforms payload data using the consuming application's transformation service and configuration.
+   * @param payload The payload to transform.
+   * @returns The transformed payload.
+   */
+  protected abstract transformPayload<T>(payload: T): T;
+
```
Comment (iamfrankiemoran) [inline projects/opal-frontend-common/components/abstract/abstract-creditor-details-base/abstract-creditor-details-base.component.ts]: [P2] The base class now requires a fairly large set of abstract adapter methods (`getPermission`, `compareVersion`, `getBusinessUnitId`, `getRefreshAccountId`, store message/mismatch setters, and `transformPayload`). That suggests the abstraction boundary is still a bit leaky. Could we group the store/permission hooks behind a smaller adapter interface or keep some of this in the consuming app’s intermediate base class? It would make the common-lib API easier to understand and harder to misuse.

File: projects/opal-frontend-common/components/abstract/abstract-creditor-details-base/abstract-creditor-details-base.component.ts [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+   * Clears the consuming account store's success message.
+   */
+  protected abstract clearSuccessMessage(): void;
+
+  /**
+   * Transforms payload data using the consuming application's transformation service and configuration.
+   * @param payload The payload to transform.
+   * @returns The transformed payload.
+   */
+  protected abstract transformPayload<T>(payload: T): T;
```
Comment (iamfrankiemoran) [inline projects/opal-frontend-common/components/abstract/abstract-creditor-details-base/abstract-creditor-details-base.component.ts]: [P2] `transformPayload<T>(payload: T): T` assumes the API payload and view payload always have exactly the same shape. That is true for the current extraction, but it may become limiting if a consumer needs to map from an API DTO into a different display model. Is that an intentional constraint? If not, it may be worth allowing separate header/tab view types in the generic parameters.

File: projects/opal-frontend-common/components/abstract/abstract-creditor-details-base/abstract-creditor-details-base.component.ts [hunk 5]
```diff
@@ -0,0 +1,137 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { Observable } from 'rxjs';
+
+import { AbstractAccountSummaryBaseComponent } from '@hmcts/opal-frontend-common/components/abstract/abstract-account-summary-base';
+
+@Component({
+  template: '',
+})
+export abstract class AbstractCreditorDetailsBaseComponent<THeader, TTab extends { version: string | null }>
```
Comment (iamfrankiemoran) [inline projects/opal-frontend-common/components/abstract/abstract-creditor-details-base/abstract-creditor-details-base.component.ts:23]: [P1] This looks like it was extracted from the `opal-frontend` creditor details base, but the PR does not show the consumer actually extending this class. Without that, it is hard to tell whether the abstraction is complete or whether the consuming app will still need a near-duplicate intermediate base. Could we either update the consumer branch to extend this class, or add a concrete usage example/README showing how the existing `FinesAccCreditorDetailsBaseComponent` maps onto these hooks?

**Review by iamfrankiemoran — CHANGES_REQUESTED** [review]
I think this extraction is going in the right direction, but I’d like us to use `AbstractAccountSummaryBaseComponent` as the model for the abstraction boundary.

That existing base keeps the shared logic generic: lifecycle, tab refresh/version handling, permissions, teardown, and header refresh. It asks consumers to provide the app-specific pieces through focused hooks such as `getHeaderDataFromRoute`, `getHeaderData`, `transformHeaderForStore`, and `transformHeaderForView`.

By comparison, `AbstractCreditorDetailsBaseComponent` currently bakes in more of the consuming app’s workflow: the `accountId` route param name, route data lookup by key, default tab handling, success banner state, version mismatch state, permission-key mapping, and store cleanup. That makes it useful for the current `opal-frontend` creditor pages, but less obviously reusable as a common-lib abstraction.

Could we tighten this so it follows the same pattern as `AbstractAccountSummaryBaseComponent`: keep only the generic creditor/account-summary orchestration in the common class, and push route/store/banner specifics into the consuming adapter class or explicit hooks? That would make the API easier to reuse and reduce the chance that future consumers have to work around assumptions from the original `opal-frontend` implementation.

File: projects/opal-frontend-common/components/abstract/abstract-creditor-details-base/abstract-creditor-details-base.component.ts (hunk 1 above)
Comment (iamfrankiemoran) [reply projects/opal-frontend-common/components/abstract/abstract-creditor-details-base/abstract-creditor-details-base.component.ts]: [P1] This still hard-codes the common base to `paramMap.get('accountId')`. It works for the current fines account routes, but it makes the common-lib abstraction depend on one consumer’s route shape and silently produces `NaN` for any different param. Please follow the pattern of `AbstractAccountSummaryBaseComponent` and make this consumer-provided, e.g. `protected getAccountIdFromRoute(): number`, or make the route param key configurable.

**Review by iamfrankiemoran — CHANGES_REQUESTED** [review]
[P2] Could we add a short README for this new abstract component, similar to `abstract-account-summary-base/readme.md`? This class is intended to be extended by consuming apps, so it would help to document the required hooks, expected store/transformer shape, route data assumptions, and a small example of how `opal-frontend` should wire it up.

[P2] New-code coverage is currently around 97.1%. That is decent, but because this is a shared abstract base used by consuming apps, could we review whether the missing lines cover important edge cases? In particular, I’d expect tests around missing/unknown permissions, route fragment fallback, refresh behaviour, teardown, and any configurable account-id route handling once added.

**Review by iamfrankiemoran — APPROVED** [review]
LGTM

**Review by ChrisGarratt122 — APPROVED** [review]
LGTM
